metrics.go

```diff
@@ -0,0 +1,217 @@
+package cke
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	v3 "github.com/coreos/etcd/clientv3"
+	"github.com/cybozu-go/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type logger struct{}
+
+func (l logger) Println(v ...interface{}) {
+	log.Error(fmt.Sprint(v...), nil)
+}
+
+const (
+	namespace     = "cke"
+	scrapeTimeout = time.Second * 10
```

Comment:
The default scrape timeout value of Prometheus is also 10 seconds.
Considering that, this value should be shortened, shouldn't this?

Reply:
Absolutely, it should be shortened.
The purpose I applied this timeout was to reclaim resources as soon as possible *after* the scraping timeout.  Now I'm reminded that the collector should return as many metrics as possible even when some collection goroutine is delayed.  The shorter timeout will enable partial responses.